intr.S

```diff
@@ -0,0 +1,91 @@
+    .set noreorder // Disable automatic instruction reordering
+
+    /* Interrupts vector */
+    .section .exception
+    .globl _interrupt_handler
+    .org 0x200
+_interrupt_handler:
```

Comment:
My implementation was mildly inspired by [Sergie's LiteBSD](https://github.com/sergev/LiteBSD/tree/master/sys/mips/pic32), and I made sure the information set that gets stored is correct. The routines that you have linked to store much more information than it is actually needed, and I wanted to avoid redundancy.